server/src/test/java/org/opensearch/common/settings/SettingTests.java

```diff
@@ -323,6 +325,27 @@ public void testValidator() {
         assertTrue(FooBarValidator.invokedWithDependencies);
     }
 
+    public void testRegexValidator() throws Exception {
+        // A regex that matches one or more digits
+        String expectedRegex = "\\d+";
+        Pattern expectedPattern = Pattern.compile(expectedRegex);
+        RegexValidator regexValidator = new RegexValidator(expectedRegex);
+
+        // Test that the pattern is correctly initialized
+        assertNotNull(expectedPattern);
+        assertNotNull(regexValidator.getPattern());
+        assertEquals(expectedPattern.toString(), regexValidator.getPattern().toString());
```

Comment:
Use `expectedPattern.pattern()` rather than the `toString()`.  They may be the same thing, but the `toString()` is intended to be human-readable while `pattern()` explicitly returns the stored string.